plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/LoggingInvoker.java

```diff
@@ -137,6 +138,18 @@ private void maybeRecord(Class<?> clazz, Supplier<String> message) {
         Class<?> clazz = classOf(lhs);
         maybeRecord(clazz, () -> clazz.getName() + "." + name);
         delegate.setProperty(lhs, name, value);
+        var g = CpsThreadGroup.current();
+        if (g != null) {
+            var e = g.getExecution();
+            if (e != null) {
+                e.getOwner().getListener().getLogger().println(Messages.LoggingInvoker_field_set(findOwner(lhs).getClass().getName(), name, value.getClass().getName()));
```

Comment:
Are you intentionally warning about _all_ property sets? It seems like for the scenarios being described it would make sense to restrict this to only apply when the receiver is `WorkflowScript` or `DSL` or whatever the right class is.

Reply:
Yeah that is what I am trying to figure out how to distinguish.